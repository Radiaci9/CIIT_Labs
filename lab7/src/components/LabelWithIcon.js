import React from 'react'
import PropTypes from 'prop-types'

export default class LabelWithIcon extends React.Component {
  
static propTypes = {
  iconSrc: PropTypes.string.isRequired,
  link: PropTypes.bool.isRequired
}
  render() {
    const { iconSrc, label, link } = this.props
    return (
      <div style={{display: 'flex', marginTop: '5px'}}>
          <img src={iconSrc} alt=""/>
          {
            !link
            ? <p name="pLink" style={{margin: '0 0 0 10px'}}>{label ? label : 'none'}</p>
            : <a name="aLink" style={{margin: '0 0 0 10px'}} href={label ? label : '#'}>{label ? label : 'none'}</a>
          }
      </div>
    )
  }
}

