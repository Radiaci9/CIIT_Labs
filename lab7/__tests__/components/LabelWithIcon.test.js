import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LabelWithIcon from '../../src/components/LabelWithIcon'

describe('<LabelWithIcon />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<LabelWithIcon iconSrc='../../img/link.svg' label='label' link={true}/>)  
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('render()', () => {
    test('renders the component without link and null label', () => {
      const wrapper = shallow(<LabelWithIcon iconSrc='../../img/link.svg' label={null} link={false}/>)  
      expect(wrapper.find('[name="pLink"]').text()).toEqual('none');
    })
  })

  describe('render()', () => {
    test('renders the component without link', () => {
      const wrapper = shallow(<LabelWithIcon iconSrc='../../img/link.svg' label='label' link={false}/>)  
      expect(wrapper.find('[name="pLink"]').text()).toEqual('label');
    })
  })

  describe('render()', () => {
    test('renders the component with link and null label', () => {
      const wrapper = shallow(<LabelWithIcon iconSrc='../../img/link.svg' label={null} link={true}/>)  
      expect(wrapper.find('[name="aLink"]').text()).toEqual('none');
    })
  })
})
