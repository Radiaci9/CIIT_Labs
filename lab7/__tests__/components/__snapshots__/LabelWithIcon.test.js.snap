// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelWithIcon /> render() renders the component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginTop": "5px",
    }
  }
>
  <img
    alt=""
    src="../../img/link.svg"
  />
  <a
    href="label"
    name="aLink"
    style={
      Object {
        "margin": "0 0 0 10px",
      }
    }
  >
    label
  </a>
</div>
`;
